tests/testAutopas/tests/pairwiseFunctors/LJFunctorTestGlobals.cpp

```diff
@@ -0,0 +1,382 @@
+  functor.SoAExtractor(cell2, cell2._particleSoABuffer);
+
+  functor.endTraversal(newton3);
+
+  double upot = functor.getUpot();
+  double virial = functor.getVirial();
+
+  EXPECT_NEAR(upot, whereFactor * expectedEnergy, absDelta)
+      << "where: " << where_str << ", newton3: " << newton3 << ", duplicatedCalculation:" << duplicatedCalculation
+      << ", interactionType: " << (interactionType == pair ? "pair" : (interactionType == own ? "own" : "verlet"))
+      << ", additionalVerletDummyParticles: " << additionalParticlesToVerletNumber
+      << ", cellWiseOwnedState: " << cellWiseOwnedState << ", numParticleReplicas: " << numParticleReplicas;
+  EXPECT_NEAR(virial, whereFactor * expectedVirial, absDelta)
+      << "where: " << where_str << ", newton3: " << newton3 << ", duplicatedCalculation:" << duplicatedCalculation
+      << ", interactionType: " << (interactionType == pair ? "pair" : (interactionType == own ? "own" : "verlet"))
+      << ", additionalVerletDummyParticles: " << additionalParticlesToVerletNumber
+      << ", cellWiseOwnedState: " << cellWiseOwnedState << ", numParticleReplicas: " << numParticleReplicas;
+}
+
+TYPED_TEST_P(LJFunctorTestGlobals, testSoAFunctorGlobalsOwn) {
+  using FuncType = TypeParam;
+  using TestType = LJFunctorTestGlobals<FuncType>;
+
+  for (bool duplicatedCalculation : {false, true}) {
+    // the own functor can only be called for inner or outside pairs! (if two particles lie in one cell they can be
+    // either both inside the process or neither of them is)
+    for (typename TestType::where_type where : {TestType::inside, TestType::outside}) {
+      for (bool newton3 : {false, true}) {
+        for (bool cellWiseOwnedState : {false, true}) {
+          for (uint64_t numParticleReplicas : {1, 2}) {
+            if (where == TestType::outside && not duplicatedCalculation) {
+              // this case does not happen and the test is not made to check this, i.e., it will fail.
+              continue;
+            }
+            this->testSoAGlobals(where, newton3, duplicatedCalculation, TestType::own, 0, cellWiseOwnedState,
+                                 numParticleReplicas);
+          }
+        }
+      }
+    }
+  }
+}
+
+TYPED_TEST_P(LJFunctorTestGlobals, testSoAFunctorGlobalsVerlet) {
+  using FuncType = TypeParam;
+  using TestType = LJFunctorTestGlobals<FuncType>;
+
+  for (size_t additionalDummyParticles = 0; additionalDummyParticles < 30; additionalDummyParticles += 5) {
+    for (bool duplicatedCalculation : {false, true}) {
+      // the own functor can only be called for inner or outside pairs! (if two particles lie in one cell they can be
+      // either both inside the process or neither of them is)
+      for (typename TestType::where_type where : {TestType::inside, TestType::boundary, TestType::outside}) {
+        for (bool newton3 : {false, true}) {
+          for (bool cellWiseOwnedState : {false, true}) {
+            for (uint64_t numParticleReplicas : {1, 2}) {
```

Comment:
I'd like it if these for loops were also all generated, i.e., that there is a generated test for each different value of these parameters...

Reply:
Not easily possible:
https://stackoverflow.com/questions/8507385/google-test-is-there-a-way-to-combine-a-test-which-is-both-type-parameterized-a